Tooltips for controls

meandavejustice opened the pull request:
- added tooltips to all controls in min-vid
- added react-tooltip dependency
- fresh build of addon
- fixes #66 

post-review:
* reworked ReactTooltip to use `data-tip` to simplify code as @pdehaan suggested
* added `classnames` dep for simplifying `hidden` class logic, thanks @pdehaan 
* fixed issue where pause tooltip was displaying as "play"
* fixed weird minimized control panel css issue
* filed https://github.com/meandavejustice/min-vid/issues/110 for issue with tooltip placement when minimized
* fixed issue of tooltips being cut off by window when maximized

pdehaan commented on app.js:
```diff
@@ -102,7 +103,8 @@ const LoadingView = React.createClass({
     return (
         <div className={'loading'} onMouseEnter={this.enterView} onMouseLeave={this.leaveView}>
-          <a onClick={this.close} className={'close ' + this.state.hovered ? '': 'hidden'} />
+        <a onClick={this.close} data-tip data-for={'close-tip'} className={'close ' + this.state.hovered ? '': 'hidden'} />
```
In [**activity-stream**](https://github.com/mozilla/activity-stream/blob/91763bb48083be73ea910fc9e02a7771e9ef6b21/content-src/components/Loader/Loader.js#L12) we use a module named [**classnames**](http://npm.im/classnames) which may be useful (or add unnecessary bloat, who knows).

```js
const classNames = require('classnames');

<a onClick={this.close}
   data-tip
   data-for={'close-tip'}
   className={classNames('close', {hidden: this.state.hovered})} />
```

... But cuts down on a bunch of inline ifs.

pdehaan commented:
Play and Pause have the same tooltips.

![youtube](https://cloud.githubusercontent.com/assets/557895/17159382/a0d16c4e-5351-11e6-8ebd-b285a650e245.png)

![youtube](https://cloud.githubusercontent.com/assets/557895/17159388/b2e2d7ec-5351-11e6-855c-526483943950.png)

pdehaan commented:
![youtube](https://cloud.githubusercontent.com/assets/557895/17159639/8fc16b6e-5353-11e6-9ff1-af480713985c.png)
Only the "Play" and "Close" buttons have tooltips in the minimized version.

pdehaan commented:
This is a pretty cool tooltip module!

It looks like we may be able to hack it where we define the tooltips inline as an attribute, and then can get away with using a single `<ReactTooltip place="bottom" effect="solid" />` instance as the defaults. Of course maybe I'm completely wrong, but it Worked For Me(tm) locally.

Also removed a bit of excess binding around string attributes since it seems to work fine without it.

```js
   <div className={'controls ' + (this.state.hovered ? '' : 'hidden ') + (this.props.minimized ? 'minimized' : '')}
             onMouseEnter={this.enterControls} onMouseLeave={this.leaveControls}>
          <div className='left'>
            <ReactTooltip place='bottom' effect='solid' />

            <a onClick={this.play} data-tip='Play'
               className={this.props.playing ? 'play hidden' : 'play'} />
            <a onClick={this.pause} data-tip='Pause'
               className={this.props.playing ? 'pause' : 'pause hidden'} />
            <a onClick={this.mute} data-tip='Mute'
               className={!this.props.muted ? 'mute' : 'mute hidden'} />
            <a onClick={this.unmute} data-tip='Unmute'
               className={this.props.muted ? 'unmute' : 'unmute hidden'} />
            <input type='range' className={this.state.showVolume ? 'volume' : 'volume hidden'}
                   min='0' max='1' step='.01' value={this.props.volume} onChange={this.setVolume}/>
          </div>

          <div className='right'>
            <a onClick={this.sendToTab} data-tip='Send to tab' className='tab' />
            <a onClick={this.minimize} data-tip='Minimize'
               className={!this.props.minimized ? 'minimize' : 'minimize hidden'} />
            <a onClick={this.maximize} data-tip='Maximize'
               className={this.props.minimized ? 'maximize' : 'maximize hidden'} />
            <a onClick={this.close} data-tip='Close' className='close' />
          </div>
        </div>
```

Possibly a few other references I overlooked, so you'll probably want to do a Ctrl+F for `data-for=` locally if the above looks kosher (and isn't full of stupid Peter typos).

6a68 commented:
This looks great overall.

I think it would be nice to eventually move to a one-tooltip model, as @pdehaan mentioned, since the user will only ever hover a single element at a time. I'd suggest we file a followup bug for that, though, rather than keep this PR open any longer.

Could the tooltip be positioned to the left or right for the controls at the corners of the player window? The 'play' and 'close' tooltips are cut off for me, because they are positioned at bottom:

![](https://www.dropbox.com/s/yj8pwy0h0kdsgr9/Screen%20Shot%202016-07-27%20at%201.06.33%20PM.png?dl=0&raw=true)

![](https://www.dropbox.com/s/2szt06tq05gmysa/Screen%20Shot%202016-07-27%20at%201.06.12%20PM.png?dl=0&raw=true)

Again, I think it'd be better to file followup bugs to position the tooltips off to one side. That might even be a Good First Bug for a new contributor :-)

I'm hitting some kind of error which is stopping me from testing further, but let's land this now and see what happens next. @meandavejustice should I merge, or do you want to enforce any kind of ordering in these PR merges?

6a68 commented:
Hmm, positioning the tooltips off to one side obscures the other controls and makes it hard to slide your mouse from control to control. Definitely file a bug and we can ask UX to figure out the best available option.

meandavejustice commented:
* reworked ReactTooltip to use `data-tip` to simplify code as @pdehaan suggested
* added `classnames` dep for simplifying `hidden` class logic, thanks @pdehaan 
* fixed issue where pause tooltip was displaying as "play"
* fixed weird minimized control panel css issue
* filed https://github.com/meandavejustice/min-vid/issues/110 for issue with tooltip placement when minimized
* fixed issue of tooltips being cut off by window when maximized
